share/functions/__fish_toggle_comment_commandline.fish

```diff
@@ -4,12 +4,13 @@
 # retrieving the command from the shell history and removing the comment chars.
 #
 # This deliberately does not execute the command when removing the comment characters to give you an
-# opportunity to modify the command.
+# opportunity to modify the command. Also if the commandline is empty, the most recently commented
+# out history item is uncommented and presented.
 
 function __fish_toggle_comment_commandline --description 'Comment/uncomment the current command'
     set -l cmdlines (commandline -b)
     if test "$cmdlines" = ""
-        return
+        set cmdlines (history search -p "#" -z | read -z)
```

Comment:
When you're doing `read`, you can just give that a variable name:

```fish
history search -p "#" -z | read -z cmdlines
```

Reply:
Ah right, thanks for pointing that out :)